src/wallet/scriptpubkeyman.cpp

```diff
@@ -384,7 +384,7 @@ bool LegacyScriptPubKeyMan::Upgrade(int prev_version, std::string& error)
         hd_upgrade = true;
     }
     // Upgrade to HD chain split if necessary
-    if (m_storage.CanSupportFeature(FEATURE_HD_SPLIT)) {
+    if (m_storage.CanSupportFeature(FEATURE_HD_SPLIT) && hdChain.nVersion < 2 /* VERSION_HD_CHAIN_SPLIT */) {
```

Comment:
any way to actually use the constant rather than referring to it and hoping the value never changes?

Reply:
Yes. the proper header is being imported now, so no need for this hack. Changed it.

Also, the value should never change because its persisted to wallet files and changing it would break backwards compatibility.